### ISSUE TITLE
    UnexpectedError: Could not locate a valid AppDelegate at root: 

### ISSUE THREAD

**Issue by shailesh-galgal** [expo/expo-cli#4542]
### Summary

when we do 
expo run:ios its shows

```
✔ Push Notifications setup for galgal-app: money.galgal.galgal
    UnexpectedError: Could not locate a valid AppDelegate at root: 
    "/Users/shaileshwadekar/Desktop/galgal/frontend"
    Please report this as an issue on 
    https://github.com/expo/expo-cli/issues
```

### Environment

when we do 
expo run:ios its shows
`    UnexpectedError: Could not locate a valid AppDelegate at root: `

### Please specify your device/emulator/simulator platform, model and version

Emulator iphon 13

### Error output

✔ Push Notifications setup for galgal-app: money.galgal.galgal
    UnexpectedError: Could not locate a valid AppDelegate at root: 
    "/Users/shaileshwadekar/Desktop/galgal/frontend"
    Please report this as an issue on 
    https://github.com/expo/expo-cli/issues

### Reproducible demo or steps to reproduce from a blank project

✔ Push Notifications setup for galgal-app: money.galgal.galgal
    UnexpectedError: Could not locate a valid AppDelegate at root: 
    "/Users/shaileshwadekar/Desktop/galgal/frontend"
    Please report this as an issue on 
    https://github.com/expo/expo-cli/issues

**Comment by andrewminton** [comment]
Same issue here with latest cli build:
UnexpectedError: Could not locate a valid AppDelegate at root

I was running this command to process a build: eas build --profile development --platform ios

Same with any other method though tbh. expo run:ios etc.

**Comment by brentvatne** [comment]
@andrewminton - do you have an ios directory in your project that does not contain a full working ios project

**Comment by andrewminton** [comment]
Nope it's all complete. pod install flushed and functioning. What I did notice just now though is the AppDelegate.m file inside the ios folder has 2 'mm' as the file extension... 

So my file looks like this /src/project/AppDelegate.mm' 

If I change the file back to AppDelegate.m the build works. 

I got a feeling somewhere in the overwriting expo-cli code, it's not accounting for an existing string filename maybe? 

@brentvatne
